src/Xamarin.Android.Tools.Bytecode/Tests/JavaAnnotation.xml

```diff
@@ -16,6 +16,18 @@
         name="java.lang.annotation.Annotation"
         name-generic-aware="java.lang.annotation.Annotation"
         jni-type="Ljava/lang/annotation/Annotation;" />
+      <runtimeVisibleAnnotations>
+        <annotation
```

Comment:
Part of the question here is twofold:

 1. What type of XML output do we want to *emit*, and
 2. What type of XML output will be easiest for `generator` to *consume*?

While `<annotation type="..."><value key=="..." /></annotation>` is easy to *emit*, I doubt that this is at all easy to *consume*.  It's useful for human reading of output, but I doubt it's useful for machine reading of output.

I suspect what we might instead want is:

```xml
<runtimeVisibleAnnotations>
  <annotation jni-type="Ljava/lang/annotation/Target;">
    <array>
      <enum jni-type="Ljava/lang/annotation/ElementType;" member="TYPE" />
      <enum jni-type="Ljava/lang/annotation/ElementType;" member="FIELD" />
      <enum jni-type="Ljava/lang/annotation/ElementType;" member="CONSTRUCTOR" />
      <enum jni-type="Ljava/lang/annotation/ElementType;" member="METHOD" />
      <enum jni-type="Ljava/lang/annotation/ElementType;" member="PARAMETER" />
      <enum jni-type="Ljava/lang/annotation/ElementType;" member="LOCAL_VARIABLE" />
    </array>
  </annotation>
  <annotation jni-type="Ljava/lang/annotation/Retention;">
    <enum jni-type="Ljava/lang/annotation/RetentionPolicy;" member="SOURCE" />
  </annotation>
</runtimeVisibleAnnotations>
```

I can imagine this being far more understandable to `generator` -- and maybe even humans? -- than the mishmash of what I came up with in PR #467, though it also means an associated increase in "bloat"/XML size as well as implementation complexity.  (Which is why I didn't do it in #467, but *also* why I didn't want to *merge* #467.)

Reply:
I am beginning to think we should stick to Agile principles here and not write code that we don't need yet, since we don't know if we'll ever need it or what shape would be best if we do.

Reply:
That's an excellent point.  In that case, do we actually need a separate `<value/>` element when the `//value/@key` attribute is always `value`?  Why not just merge into `<annotation/>`?

```xml
<annotation
  type="Ljava/lang/annotation/Target;"
  elements="[Enum(Ljava/lang/annotation/ElementType;.TYPE), Enum(Ljava/lang/annotation/ElementType;.FIELD), Enum(Ljava/lang/annotation/ElementType;.CONSTRUCTOR), Enum(Ljava/lang/annotation/ElementType;.METHOD), Enum(Ljava/lang/annotation/ElementType;.PARAMETER), Enum(Ljava/lang/annotation/ElementType;.LOCAL_VARIABLE)]"
/>
```

Reply:
I'm not sure I understand, the keys have different names which should be preserved:

```
<annotation type="Lkotlin/Metadata;">
    <value key="mv">[1, 1, 15]</value>
    <value key="bv">[1, 0, 3]</value>
</annotation>
```

Reply:
I'm an idiot and misread this line: https://github.com/xamarin/java.interop/pull/502/files#diff-759d8653993a49e7976b261c1ef45a72R335

```csharp
return new XElement ("value",
```

I misread that as the attribute value.  Doh!